README.md

```diff
@@ -21,6 +21,36 @@ pip install .[all]  # Install the remaining elements of this package.
 
 Aside from the base packages, you can optionally install `dev` tools using this tag, `materials` dependencies using this tag, or `molecules` packages using this tag. The simplest option is to use the `all` tag, as above, which installs all dependencies.
 
+## Main Components of the GFlowNet Library
+
+The GFlowNet library has 5 core components, each playing a crucial role in the network's operation. Understanding these components is essential for effectively using and extending the library for your own task. These 5 component are the Logger, Proxy, Environment,  Policies (Forward and Backward), and GFlowNet Agent. 
+
+### Logger
+The purpose of the Logger is to manage all logging activities during the training and evaluation of the network. It captures and stores logs to track the model's performance and debugging information. For instance, it logs details such as training progress, performance metrics, and any potential errors or warnings that occur. It also integrates to WandB providing a cloud-based platform for visualizing and comparing experiments. 
+
+### Environment
+The Environment is main and most important component of GFlowNet Library. To understand the Environment, let us consider a simple enviroment currently implemented in the library, the Scramble enviroment. 
+
+The Scramble environment simulates a simple letter arrangement game where sequences are constructed by adding one letter at a time, up to a maximum sequence length (in our case 7). Each enviroment would have State Representation and Actions. For instance, for scramble enviroment, Each ``State`` within the environment is a list of indices corresponding to letters. These indices start from 1 and are padded with index 0 to denote unused slots up to the maximum length. For example, if our sequence length is 7, and our constructed word is `Alex`, we will have `[1, 11, 4, 23, 0, 0, 0]`. In the library, we already have helper functions that automatically format and convert the states to human readable and vice versa. 
```

Comment:
Note that it is ["Scrabble"](https://en.wikipedia.org/wiki/Scrabble), not "Scramble" ;)

Reply:
Good catch thanks :)